provider/resource_keycloak_openid_client_service_account_role.go

```diff
@@ -90,23 +116,27 @@ func resourceKeycloakOpenidClientServiceAccountRoleRead(data *schema.ResourceDat
 func resourceKeycloakOpenidClientServiceAccountRoleDelete(data *schema.ResourceData, meta interface{}) error {
 	keycloakClient := meta.(*keycloak.KeycloakClient)
 
-	realmId := data.Get("realm_id").(string)
-	serviceAccountUserId := data.Get("service_account_user_id").(string)
-	clientId := data.Get("client_id").(string)
-	id := data.Id()
+	serviceAccountRole, err := getOpenidClientServiceAccountRoleFromData(data, keycloakClient)
+	if err != nil {
+		return err
+	}
 
-	return keycloakClient.DeleteOpenidClientServiceAccountRole(realmId, serviceAccountUserId, clientId, id)
+	if serviceAccountRole != nil {
```

Comment:
Same comment as above, I don't think this can ever be false

Reply:
Had another problem. And tested locally this change, forgot to remove